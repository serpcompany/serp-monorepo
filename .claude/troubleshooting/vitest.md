https://vitest.dev/guide/common-errors.html