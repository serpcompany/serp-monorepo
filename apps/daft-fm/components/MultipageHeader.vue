<template>
  <div>
    <header
      ref="header"
      :class="{
        'py-2': isScrolled,
        'py-4 sm:py-8': !isScrolled,
        'fixed top-0 left-0 z-50 w-full bg-[var(--ui-bg)]': isScrolled
      }"
    >
      <div class="mx-auto max-w-6xl px-4 sm:px-6 lg:px-8">
        <!-- header top -->
        <div class="flex flex-col items-center justify-between sm:flex-row">
          <div
            class="flex w-full items-center justify-between sm:justify-start"
          >
            <!-- image -->
            <!-- <img

                 :alt="`${item.name} logo`"
                 class="object-contain transition-all duration-300"
                 :class="{ 'h-10': isScrolled, 'h-16': !isScrolled }" /> -->

            <!-- info -->
            <div
              class="ml-4 flex-grow"
              :class="{ 'hidden sm:block': isScrolled }"
            >
              <h1
                class="text-primary text-2xl font-bold transition-all duration-300 sm:text-3xl"
                :class="{
                  'text-xl sm:text-lg': isScrolled,
                  'mt-0': isScrolled
                }"
              >
                {{ name }}
              </h1>
            </div>
          </div>
        </div>
        <!-- header bottom (nav links) -->
        <div
          class="-mx-4 mt-4 overflow-x-auto px-4 sm:px-0"
          :class="{ 'mt-2': isScrolled }"
        >
          <nav
            aria-label="page sections"
            class="flex space-x-4 border-t border-b whitespace-nowrap"
          >
            <nuxt-link
              v-for="section in sections"
              :key="section"
              :href="'#' + section.toLowerCase()"
              class="sectionLinks text-muted-foreground hover:text-primary flex-shrink-0 px-1 py-2 text-sm transition-colors duration-200 sm:px-2 sm:text-base"
            >
              {{ section }}
            </nuxt-link>
          </nav>
        </div>
      </div>
    </header>
  </div>
</template>

<script setup lang="ts">
import { ref, onMounted, onUnmounted } from 'vue';

const props = defineProps({
  name: {
    type: String,
    required: true
  },
  image: {
    type: String,
    required: false
  },
  sections: {
    type: Array as () => string[],
    required: true
  }
});

const header = ref(null);
const isScrolled = ref(false);

const handleScroll = () => {
  isScrolled.value = window.scrollY > 0;
};

onMounted(() => {
  window.addEventListener('scroll', handleScroll);
});

onUnmounted(() => {
  window.removeEventListener('scroll', handleScroll);
});
</script>
