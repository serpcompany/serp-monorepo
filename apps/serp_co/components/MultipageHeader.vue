<template>
  <div>
    <header
      ref="header"
      class="bg-[var(--ui-bg)]"
      :class="{
        'py-2': isScrolled,
        'py-2 sm:py-2': !isScrolled,
        'fixed top-0 left-0 z-50 w-full': isScrolled
      }"
    >
      <div class="mx-auto max-w-6xl sm:px-6 lg:px-8">
        <!-- header top -->

        <div class="flex flex-col items-center justify-between sm:flex-row">
          <!-- image -->
          <div class="flex w-full items-start justify-between">
            <nuxt-img
              v-if="image"
              :src="image"
              :alt="`${name} logo`"
              class="mr-2 object-contain transition-all duration-300"
              :class="{ 'h-10': isScrolled, 'h-16': !isScrolled }"
            />

            <!-- name -->
            <div class="flex-grow" :class="{ 'hidden sm:block': isScrolled }">
              <h1
                class="text-primary text-xl font-bold transition-all duration-300 sm:text-3xl"
                :class="{
                  'text-xl sm:text-lg': isScrolled,
                  'mt-0': isScrolled
                }"
              >
                {{ name }}
              </h1>
              <span class="text-xs">{{ oneLiner }}</span>
            </div>
          </div>

          <!-- visit website button -->
          <div class="sm:flex-end flex w-full pt-4 sm:pt-0">
            <div class="mr-auto ml-0 sm:mr-0 sm:ml-auto">
              <nuxt-link
                :to="serply_link"
                target="_blank"
                class="border px-4 py-2"
                >Visit Website</nuxt-link
              >
            </div>
          </div>
        </div>

        <!-- header bottom (nav links) -->
        <div
          class="-mx-4 mt-4 overflow-x-auto px-4 sm:px-0"
          :class="{ 'mt-2': isScrolled }"
        >
          <nav
            aria-label="page sections"
            class="flex space-x-4 border-t border-b whitespace-nowrap"
          >
            <nuxt-link
              v-for="section in sections"
              :key="section"
              :href="'#' + section.toLowerCase()"
              class="sectionLinks text-muted-foreground hover:text-primary flex-shrink-0 px-1 py-2 text-sm transition-colors duration-200 sm:px-2 sm:text-base"
            >
              {{ section }}
            </nuxt-link>
          </nav>
        </div>
      </div>
    </header>
  </div>
</template>

<script setup lang="ts">
defineProps({
  name: {
    type: String,
    required: true
  },
  oneLiner: {
    type: String,
    required: false
  },
  image: {
    type: String,
    required: false
  },
  sections: {
    type: Array as () => string[],
    required: true
  },
  serply_link: {
    type: String,
    required: true
  }
});

const header = ref(null);
const isScrolled = ref(false);
</script>
