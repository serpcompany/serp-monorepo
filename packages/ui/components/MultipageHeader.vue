<script setup lang="ts">
  defineProps({
    name: {
      type: String,
      required: true
    },
    oneLiner: {
      type: String,
      required: false,
      default: ''
    },
    image: {
      type: String,
      required: false,
      default: ''
    },
    sections: {
      type: Array as () => string[],
      required: true
    },
    serplyLink: {
      type: String,
      required: true
    }
  });

  const header = ref(null);
  const isScrolled = ref(false);

  // Add proper Nuxt-friendly scroll detection
  onMounted(() => {
    const handleScroll = () => {
      isScrolled.value = window.scrollY > 0;
    };

    window.addEventListener('scroll', handleScroll);
    handleScroll(); // Initial check

    onBeforeUnmount(() => {
      window.removeEventListener('scroll', handleScroll);
    });
  });
</script>

<template>
  <div>
    <header
      ref="header"
      class="w-full bg-[var(--ui-bg)] transition-all duration-300"
      :class="{
        'py-2': isScrolled,
        'py-2 sm:py-4': !isScrolled,
        'sticky top-0 left-0': isScrolled
      }"
      style="z-index: 10"
    >
      <div class="mx-auto max-w-7xl px-4 sm:px-6 lg:px-8">
        <!-- When not scrolled: show the regular layout -->
        <div v-if="!isScrolled">
          <div class="flex flex-col items-center justify-between sm:flex-row">
            <!-- Logo and title section -->
            <div class="flex w-full items-start justify-between">
              <NuxtImg
                v-if="image"
                :src="image"
                :alt="`${name} logo`"
                class="mr-2 h-16 object-contain"
              />
              <div class="flex-grow">
                <h1 class="text-xl font-bold text-(--ui-primary) sm:text-3xl">
                  {{ name }}
                </h1>
                <span class="text-xs">{{ oneLiner }}</span>
              </div>
            </div>

            <!-- Action buttons -->
            <div
              class="flex w-full flex-col items-center justify-end gap-3 pt-4 sm:flex-row sm:pt-0"
            >
              <slot name="upvote"></slot>
              <NuxtLink
                :href="serplyLink"
                target="_blank"
                class="flex w-full items-center justify-center gap-2 rounded-full bg-black px-4 py-2 font-medium text-white transition-colors hover:bg-neutral-800 sm:w-auto dark:bg-white dark:text-black dark:hover:bg-neutral-200"
              >
                Visit Website
                <svg
                  class="h-4 w-4"
                  xmlns="http://www.w3.org/2000/svg"
                  viewBox="0 0 24 24"
                  fill="none"
                  stroke="currentColor"
                  stroke-width="2"
                  stroke-linecap="round"
                  stroke-linejoin="round"
                >
                  <path
                    d="M18 13v6a2 2 0 0 1-2 2H5a2 2 0 0 1-2-2V8a2 2 0 0 1 2-2h6"
                  />
                  <polyline points="15 3 21 3 21 9" />
                  <line x1="10" y1="14" x2="21" y2="3" />
                </svg>
              </NuxtLink>
            </div>
          </div>

          <!-- Navigation for non-scrolled state -->
          <div class="-mx-4 mt-4 overflow-x-auto px-4 sm:px-0">
            <nav
              aria-label="page sections"
              class="flex space-x-4 border-t border-b whitespace-nowrap"
            >
              <NuxtLink
                v-for="section in sections"
                :key="section"
                :href="'#' + section.toLowerCase()"
                class="sectionLinks flex-shrink-0 px-1 py-2 text-sm font-medium text-(--ui-primary) transition-colors duration-200 hover:text-(--ui-primary-dark) sm:px-2 sm:text-base"
              >
                {{ section }}
              </NuxtLink>
            </nav>
          </div>
        </div>

        <!-- When scrolled: show compact layout with everything in one row -->
        <div v-else class="flex items-center justify-between">
          <!-- Left section with logo, title and nav -->
          <div class="flex items-center space-x-6">
            <!-- Logo and title -->
            <div class="flex items-center">
              <NuxtImg
                v-if="image"
                :src="image"
                :alt="`${name} logo`"
                class="mr-2 h-8 object-contain"
              />
              <h1 class="text-base font-bold text-(--ui-primary)">
                {{ name }}
              </h1>
            </div>

            <!-- Nav links left-aligned -->
            <nav
              aria-label="page sections"
              class="flex space-x-4 overflow-x-auto whitespace-nowrap"
            >
              <NuxtLink
                v-for="section in sections"
                :key="section"
                :href="'#' + section.toLowerCase()"
                class="sectionLinks flex-shrink-0 px-2 py-2 text-sm font-medium text-(--ui-primary) transition-colors duration-200 hover:text-(--ui-primary-dark)"
              >
                {{ section }}
              </NuxtLink>
            </nav>
          </div>

          <!-- CTA on the right -->
          <NuxtLink
            :href="serplyLink"
            target="_blank"
            class="flex items-center justify-center gap-1 rounded-full bg-black px-3 py-1 text-sm font-medium text-white transition-colors hover:bg-neutral-800 dark:bg-white dark:text-black dark:hover:bg-neutral-200"
          >
            Visit Website
            <svg
              class="h-3 w-3"
              xmlns="http://www.w3.org/2000/svg"
              viewBox="0 0 24 24"
              fill="none"
              stroke="currentColor"
              stroke-width="2"
              stroke-linecap="round"
              stroke-linejoin="round"
            >
              <path
                d="M18 13v6a2 2 0 0 1-2 2H5a2 2 0 0 1-2-2V8a2 2 0 0 1 2-2h6"
              />
              <polyline points="15 3 21 3 21 9" />
              <line x1="10" y1="14" x2="21" y2="3" />
            </svg>
          </NuxtLink>
        </div>
      </div>
    </header>
  </div>
</template>
