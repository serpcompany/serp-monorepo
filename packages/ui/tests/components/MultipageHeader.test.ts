import { describe, expect, it } from 'vitest';
import MultipageHeader from '../../components/MultipageHeader.vue';
import ComponentRender from '../componentRender';

describe('MultipageHeader Snapshot', () => {
  const baseProps = {
    name: 'Music Header',
    sections: ['Home', 'About', 'Contact'],
    serply_link: 'https://example.com'
  };

  it.each([
    // Default rendering with minimal required props.
    ['default rendering', { props: { ...baseProps } }],
    // With an optional oneLiner and an image provided.
    [
      'with oneLiner and image',
      {
        props: {
          ...baseProps,
          oneLiner: 'One liner test',
          image: 'https://example.com/logo.png'
        }
      }
    ],
    // Edge-case: empty sections.
    ['with empty sections', { props: { ...baseProps, sections: [] } }],
    // Rendering with a custom upvote slot.
    [
      'with upvote slot',
      {
        props: { ...baseProps },
        slots: { upvote: () => '<button>Upvote</button>' }
      }
    ]
  ])(
    'renders %s correctly',
    async (desc: string, options: { props: unknown; slots?: unknown }) => {
      const html = await ComponentRender(
        `MultipageHeader ${desc}`,
        options,
        MultipageHeader
      );
      expect(html).toMatchSnapshot();
    }
  );
});
