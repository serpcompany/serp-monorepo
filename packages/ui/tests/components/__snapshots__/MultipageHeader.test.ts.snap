// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`MultipageHeader Snapshot > renders default rendering correctly 1`] = `
"<div serply_link="https://example.com">
  <header class="w-full bg-[var(--ui-bg)] transition-all duration-300 py-2 sm:py-4" style="z-index: 0;">
    <div class="mx-auto max-w-7xl px-4 sm:px-6 lg:px-8">
      <!-- When not scrolled: show the regular layout -->
      <div>
        <div class="flex flex-col items-center justify-between sm:flex-row">
          <!-- Logo and title section -->
          <div class="flex w-full items-start justify-between">
            <!--v-if-->
            <!-- Use flex-grow to allow this column to expand -->
            <div class="flex-grow">
              <h1 class="text-xl font-bold text-gray-900 sm:text-3xl dark:text-gray-100">Music Header</h1>
              <!-- <span class="text-xs text-gray-700 dark:text-gray-300">{{
                  oneLiner
                }}</span> -->
            </div>
          </div><!-- Action buttons -->
          <!-- Change w-full to w-auto or remove width entirely so it only takes up the space it needs -->
          <div class="flex flex-none flex-col items-center justify-end gap-3 pt-4 sm:flex-row sm:pt-0"><a rel="noopener noreferrer" target="_blank" class="flex h-[42px] w-auto items-center justify-center gap-2 rounded-md bg-black px-4 py-2 font-medium text-white transition-colors hover:bg-neutral-800 sm:w-auto dark:bg-white dark:text-black dark:hover:bg-neutral-200"> Visit Website <svg class="h-4 w-4" xmlns="http://www.w3.org/2000/svg" viewBox="0 0 24 24" fill="none" stroke="currentColor" stroke-width="2" stroke-linecap="round" stroke-linejoin="round">
                <path d="M18 13v6a2 2 0 0 1-2 2H5a2 2 0 0 1-2-2V8a2 2 0 0 1 2-2h6"></path>
                <polyline points="15 3 21 3 21 9"></polyline>
                <line x1="10" y1="14" x2="21" y2="3"></line>
              </svg></a></div>
        </div><!-- Navigation for non-scrolled state -->
        <div class="-mx-4 mt-4 overflow-x-auto px-4 sm:px-0">
          <nav aria-label="page sections" class="flex space-x-4 whitespace-nowrap"><a href="#home" class="sectionLinks flex-shrink-0 rounded-sm px-1 py-2 text-sm font-medium text-gray-900 transition-colors duration-200 hover:bg-gray-50 hover:text-gray-600 sm:px-2 sm:text-base dark:text-gray-100 dark:hover:bg-gray-800 dark:hover:text-gray-300">Home</a><a href="#about" class="sectionLinks flex-shrink-0 rounded-sm px-1 py-2 text-sm font-medium text-gray-900 transition-colors duration-200 hover:bg-gray-50 hover:text-gray-600 sm:px-2 sm:text-base dark:text-gray-100 dark:hover:bg-gray-800 dark:hover:text-gray-300">About</a><a href="#contact" class="sectionLinks flex-shrink-0 rounded-sm px-1 py-2 text-sm font-medium text-gray-900 transition-colors duration-200 hover:bg-gray-50 hover:text-gray-600 sm:px-2 sm:text-base dark:text-gray-100 dark:hover:bg-gray-800 dark:hover:text-gray-300">Contact</a></nav>
        </div>
      </div>
    </div>
  </header>
</div>"
`;

exports[`MultipageHeader Snapshot > renders with empty sections correctly 1`] = `
"<div serply_link="https://example.com">
  <header class="w-full bg-[var(--ui-bg)] transition-all duration-300 py-2 sm:py-4" style="z-index: 0;">
    <div class="mx-auto max-w-7xl px-4 sm:px-6 lg:px-8">
      <!-- When not scrolled: show the regular layout -->
      <div>
        <div class="flex flex-col items-center justify-between sm:flex-row">
          <!-- Logo and title section -->
          <div class="flex w-full items-start justify-between">
            <!--v-if-->
            <!-- Use flex-grow to allow this column to expand -->
            <div class="flex-grow">
              <h1 class="text-xl font-bold text-gray-900 sm:text-3xl dark:text-gray-100">Music Header</h1>
              <!-- <span class="text-xs text-gray-700 dark:text-gray-300">{{
                  oneLiner
                }}</span> -->
            </div>
          </div><!-- Action buttons -->
          <!-- Change w-full to w-auto or remove width entirely so it only takes up the space it needs -->
          <div class="flex flex-none flex-col items-center justify-end gap-3 pt-4 sm:flex-row sm:pt-0"><a rel="noopener noreferrer" target="_blank" class="flex h-[42px] w-auto items-center justify-center gap-2 rounded-md bg-black px-4 py-2 font-medium text-white transition-colors hover:bg-neutral-800 sm:w-auto dark:bg-white dark:text-black dark:hover:bg-neutral-200"> Visit Website <svg class="h-4 w-4" xmlns="http://www.w3.org/2000/svg" viewBox="0 0 24 24" fill="none" stroke="currentColor" stroke-width="2" stroke-linecap="round" stroke-linejoin="round">
                <path d="M18 13v6a2 2 0 0 1-2 2H5a2 2 0 0 1-2-2V8a2 2 0 0 1 2-2h6"></path>
                <polyline points="15 3 21 3 21 9"></polyline>
                <line x1="10" y1="14" x2="21" y2="3"></line>
              </svg></a></div>
        </div><!-- Navigation for non-scrolled state -->
        <div class="-mx-4 mt-4 overflow-x-auto px-4 sm:px-0">
          <nav aria-label="page sections" class="flex space-x-4 whitespace-nowrap"></nav>
        </div>
      </div>
    </div>
  </header>
</div>"
`;

exports[`MultipageHeader Snapshot > renders with oneLiner and image correctly 1`] = `
"<div serply_link="https://example.com">
  <header class="w-full bg-[var(--ui-bg)] transition-all duration-300 py-2 sm:py-4" style="z-index: 0;">
    <div class="mx-auto max-w-7xl px-4 sm:px-6 lg:px-8">
      <!-- When not scrolled: show the regular layout -->
      <div>
        <div class="flex flex-col items-center justify-between sm:flex-row">
          <!-- Logo and title section -->
          <div class="flex w-full items-start justify-between"><img alt="Music Header logo" data-nuxt-img="" srcset="https://example.com/logo.png 1x, https://example.com/logo.png 2x" class="mr-2 h-16 object-contain" src="https://example.com/logo.png"><!-- Use flex-grow to allow this column to expand -->
            <div class="flex-grow">
              <h1 class="text-xl font-bold text-gray-900 sm:text-3xl dark:text-gray-100">Music Header</h1>
              <!-- <span class="text-xs text-gray-700 dark:text-gray-300">{{
                  oneLiner
                }}</span> -->
            </div>
          </div><!-- Action buttons -->
          <!-- Change w-full to w-auto or remove width entirely so it only takes up the space it needs -->
          <div class="flex flex-none flex-col items-center justify-end gap-3 pt-4 sm:flex-row sm:pt-0"><a rel="noopener noreferrer" target="_blank" class="flex h-[42px] w-auto items-center justify-center gap-2 rounded-md bg-black px-4 py-2 font-medium text-white transition-colors hover:bg-neutral-800 sm:w-auto dark:bg-white dark:text-black dark:hover:bg-neutral-200"> Visit Website <svg class="h-4 w-4" xmlns="http://www.w3.org/2000/svg" viewBox="0 0 24 24" fill="none" stroke="currentColor" stroke-width="2" stroke-linecap="round" stroke-linejoin="round">
                <path d="M18 13v6a2 2 0 0 1-2 2H5a2 2 0 0 1-2-2V8a2 2 0 0 1 2-2h6"></path>
                <polyline points="15 3 21 3 21 9"></polyline>
                <line x1="10" y1="14" x2="21" y2="3"></line>
              </svg></a></div>
        </div><!-- Navigation for non-scrolled state -->
        <div class="-mx-4 mt-4 overflow-x-auto px-4 sm:px-0">
          <nav aria-label="page sections" class="flex space-x-4 whitespace-nowrap"><a href="#home" class="sectionLinks flex-shrink-0 rounded-sm px-1 py-2 text-sm font-medium text-gray-900 transition-colors duration-200 hover:bg-gray-50 hover:text-gray-600 sm:px-2 sm:text-base dark:text-gray-100 dark:hover:bg-gray-800 dark:hover:text-gray-300">Home</a><a href="#about" class="sectionLinks flex-shrink-0 rounded-sm px-1 py-2 text-sm font-medium text-gray-900 transition-colors duration-200 hover:bg-gray-50 hover:text-gray-600 sm:px-2 sm:text-base dark:text-gray-100 dark:hover:bg-gray-800 dark:hover:text-gray-300">About</a><a href="#contact" class="sectionLinks flex-shrink-0 rounded-sm px-1 py-2 text-sm font-medium text-gray-900 transition-colors duration-200 hover:bg-gray-50 hover:text-gray-600 sm:px-2 sm:text-base dark:text-gray-100 dark:hover:bg-gray-800 dark:hover:text-gray-300">Contact</a></nav>
        </div>
      </div>
    </div>
  </header>
</div>"
`;

exports[`MultipageHeader Snapshot > renders with upvote slot correctly 1`] = `
"<div serply_link="https://example.com">
  <header class="w-full bg-[var(--ui-bg)] transition-all duration-300 py-2 sm:py-4" style="z-index: 0;">
    <div class="mx-auto max-w-7xl px-4 sm:px-6 lg:px-8">
      <!-- When not scrolled: show the regular layout -->
      <div>
        <div class="flex flex-col items-center justify-between sm:flex-row">
          <!-- Logo and title section -->
          <div class="flex w-full items-start justify-between">
            <!--v-if-->
            <!-- Use flex-grow to allow this column to expand -->
            <div class="flex-grow">
              <h1 class="text-xl font-bold text-gray-900 sm:text-3xl dark:text-gray-100">Music Header</h1>
              <!-- <span class="text-xs text-gray-700 dark:text-gray-300">{{
                  oneLiner
                }}</span> -->
            </div>
          </div><!-- Action buttons -->
          <!-- Change w-full to w-auto or remove width entirely so it only takes up the space it needs -->
          <div class="flex flex-none flex-col items-center justify-end gap-3 pt-4 sm:flex-row sm:pt-0">&lt;button&gt;Upvote&lt;/button&gt;<a rel="noopener noreferrer" target="_blank" class="flex h-[42px] w-auto items-center justify-center gap-2 rounded-md bg-black px-4 py-2 font-medium text-white transition-colors hover:bg-neutral-800 sm:w-auto dark:bg-white dark:text-black dark:hover:bg-neutral-200"> Visit Website <svg class="h-4 w-4" xmlns="http://www.w3.org/2000/svg" viewBox="0 0 24 24" fill="none" stroke="currentColor" stroke-width="2" stroke-linecap="round" stroke-linejoin="round">
                <path d="M18 13v6a2 2 0 0 1-2 2H5a2 2 0 0 1-2-2V8a2 2 0 0 1 2-2h6"></path>
                <polyline points="15 3 21 3 21 9"></polyline>
                <line x1="10" y1="14" x2="21" y2="3"></line>
              </svg></a></div>
        </div><!-- Navigation for non-scrolled state -->
        <div class="-mx-4 mt-4 overflow-x-auto px-4 sm:px-0">
          <nav aria-label="page sections" class="flex space-x-4 whitespace-nowrap"><a href="#home" class="sectionLinks flex-shrink-0 rounded-sm px-1 py-2 text-sm font-medium text-gray-900 transition-colors duration-200 hover:bg-gray-50 hover:text-gray-600 sm:px-2 sm:text-base dark:text-gray-100 dark:hover:bg-gray-800 dark:hover:text-gray-300">Home</a><a href="#about" class="sectionLinks flex-shrink-0 rounded-sm px-1 py-2 text-sm font-medium text-gray-900 transition-colors duration-200 hover:bg-gray-50 hover:text-gray-600 sm:px-2 sm:text-base dark:text-gray-100 dark:hover:bg-gray-800 dark:hover:text-gray-300">About</a><a href="#contact" class="sectionLinks flex-shrink-0 rounded-sm px-1 py-2 text-sm font-medium text-gray-900 transition-colors duration-200 hover:bg-gray-50 hover:text-gray-600 sm:px-2 sm:text-base dark:text-gray-100 dark:hover:bg-gray-800 dark:hover:text-gray-300">Contact</a></nav>
        </div>
      </div>
    </div>
  </header>
</div>"
`;
